apps/emqx/src/emqx_persistent_message_ds_replayer.erl

```diff
@@ -611,6 +617,34 @@ shuffle(L0) ->
     {_, L} = lists:unzip(L2),
     L.
 
+get_next_n_messages_from_group(Group, Ranges, N) ->
+    case get_next_stream_from_group(Group) of
+        {DSStream, Group2} ->
+            ItBegin = get_last_iterator(DSStream, Ranges),
+            case emqx_ds:next(?PERSISTENT_MESSAGE_DB, ItBegin, N) of
+                {ok, ItEnd, Messages} ->
+                    {ok, DSStream, ItBegin, ItEnd, Messages};
+                {ok, end_of_stream} ->
+                    %% Q: can we skip the completed stream?
```

Comment:
Yes, I think it'll be nice to ignore the completed streams. Store the RankY of the last completed stream, perhaps?